crates/sui-types/src/temporary_store.rs

```diff
@@ -689,6 +688,154 @@ impl<S> TemporaryStore<S> {
     }
 }
 
+impl<S: ObjectStore> TemporaryStore<S> {
+    /// returns lists of (objects whose owner we must authenticate, objects whose owner has already been authenticated)
+    fn get_objects_to_authenticate(
+        &self,
+        sender: &SuiAddress,
+        gas: &[ObjectRef],
+    ) -> SuiResult<(HashSet<ObjectID>, HashSet<ObjectID>)> {
+        let gas_objs: HashSet<&ObjectID> = gas.iter().map(|g| &g.0).collect();
+        let mut objs_to_authenticate = HashSet::new();
+        let mut authenticated_objs = HashSet::new();
+        for (id, obj) in &self.input_objects {
+            if gas_objs.contains(id) {
+                // gas could be owned by either the sender (common case) or sponsor (if this is a sponsored tx,
+                // which we do not know inside this function).
```

Comment:
If we also want to authenticate gas objects, we could simply call this outside of execution_engine, in authority.rs, that way we have access to sponsor as well

Reply:
Good point, will look into that.

Reply:
I checked, and the problem with this is that a lot of our tests for our authentication logic are in `sui-adapter-transactional-tests`, which uses `execution_engine` but not `authority`. Thus, my inclination is to keep calling this inside `execution_engine` for now to maximally leverage our existing tests, and perhaps we can figure out how to pass the sponsor down later if we want to simplify the code or do more stringent sponsor-y checking.